### PR TITLE
Package multiple firmware Images into single capsule

### PR DESCRIPTION
**Pull request by raghavag**
This patch will update GenCapsuleFirmware.py tool to package
multiple firmware update images into single capsule.

Example:

python GenCapsuleFirmware.py -p BIOS pld1.bin -p <GUID> pld2.bin
	   -p CSME Pld3.bin -p CSMD Pld4.bin -o img.bin -k  TestSigningPrivateKey.pem

The following are the changes made

1) Capsule header format is changed to align with
   EFI_FIRMWARE_MANAGEMENT_CAPSULE_HEADER
2) Each firmware image in the capsule will have
   EFI_FIRMWARE_MANAGEMENT_CAPSULE_IMAGE_HEADER
3) BIOS, CSME, CSME Update Driver (CSMD) and CFGData payloads
   can be passed to the tool using their names as suggested in
   example.
4) Unknown payloads can also be passed to the tool with associated
   guid as input.

Signed-off-by: Raghava Gudla <raghava.gudla@intel.com>

### CODE REVIEW

File: BootloaderCorePkg/Tools/GenCapsuleFirmware.py [hunk 1]
```diff
@@ -15,105 +15,532 @@
...
-
-	os.remove(pubkey_file)
-	os.remove(fwupdate_bin_file)
+    _fields_ = [
+        ('FileGuid', ARRAY(c_uint8, 16)), ('HeaderSize', c_uint32),
+        ('FirmwreVersion', c_uint32), ('CapsuleFlags', c_uint32),
+        ('PubKeyOffset', c_uint32), ('PubKeySize', c_uint32),
+        ('ImageOffset', c_uint32), ('ImageSize', c_uint32),
+        ('SignatureOffset', c_uint32), ('SignatureSize', c_uint32),
+        ('Reserved', ARRAY(c_uint32, 3))
```
Comment (gdong1) [inline BootloaderCorePkg/Tools/GenCapsuleFirmware.py]: Had better have one field per line as before for better readable.

File: BootloaderCorePkg/Tools/GenCapsuleFirmware.py [hunk 2]
```diff
@@ -15,105 +15,532 @@
...
+    ]
+
+
+def SignImage(RawData, OutFile, PrivKey):
```
Comment (gdong1) [inline BootloaderCorePkg/Tools/GenCapsuleFirmware.py]: It is difficult to get change difference since SignImage() was removed and then added here.
Is it caused by CR or space or other unnecessary change?

File: BootloaderCorePkg/Tools/GenCapsuleFirmware.py [hunk 3]
```diff
@@ -15,105 +15,532 @@
...
-
-  #
-  # Create command line argument parser object
-  #
-  parser = argparse.ArgumentParser()
-  parser.add_argument('-b',  '--image', dest='Image', type=str, help='Raw image into firmware update capsule image', required = True)
-  parser.add_argument('-t',  '--type', dest='Type', choices=['b','c'], help='Image type b:Bootloader c:CfgData', default = 'b')
-  parser.add_argument('-k',  '--priv_key', dest='PrivKey', type=str, help='Private RSA 2048 key in PEM format to sign image', required = True)
-  parser.add_argument('-o',  '--output', dest='NewImage', type=str, help='Output file for signed image', required = True)
-  parser.add_argument("-q", "--quiet", dest='Quiet', action="store_true", help="without output messages or temp files")
```
Comment (gdong1) [inline BootloaderCorePkg/Tools/GenCapsuleFirmware.py]: If removed the arguments, how it could support the parameter "-p", "-o" and "-k" in your commit message?

Comment (raghavag) [reply BootloaderCorePkg/Tools/GenCapsuleFirmware.py]: If you see what are removed, these are all unused args, Look from line# 513 in the new file, args are parsed there

File: BootloaderCorePkg/Tools/GenCapsuleFirmware.py (hunk 2 above)
Comment (raghavag) [reply BootloaderCorePkg/Tools/GenCapsuleFirmware.py]: I believe when I correct your above comment, this will be resolved as well.

BTW, it is not removed and added here, it is just alignment issue

File: BootloaderCorePkg/Tools/GenCapsuleFirmware.py (hunk 1 above)
Comment (raghavag) [reply BootloaderCorePkg/Tools/GenCapsuleFirmware.py]: Will Fix this

Comment (raghavag) [reply BootloaderCorePkg/Tools/GenCapsuleFirmware.py]: Fixed this in latest release

File: BootloaderCorePkg/Tools/GenCapsuleFirmware.py (hunk 3 above)
Comment (gdong1) [reply BootloaderCorePkg/Tools/GenCapsuleFirmware.py]: sorry I missed the one you added since code added/removed are not back to back in the patch.
That's why we prefer only changing necessary code in patch.  The new format is not aligned in different parameter. the old one is more readable.
